packages/js/components/src/table/index.tsx

```diff
@@ -0,0 +1,250 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { createElement, Fragment, useState } from '@wordpress/element';
+import { find, first, without } from 'lodash';
+import React from 'react';
+import {
+	Card,
+	CardBody,
+	CardFooter,
+	CardHeader,
+	// @ts-expect-error: Suppressing Module '"@wordpress/components"' has no exported member '__experimentalText'
+	__experimentalText as Text,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import EllipsisMenu from '../ellipsis-menu';
+import MenuItem from '../ellipsis-menu/menu-item';
+import MenuTitle from '../ellipsis-menu/menu-title';
+import Pagination from '../pagination';
+import Table from './table';
+import TablePlaceholder from './placeholder';
+import TableSummary, { TableSummaryPlaceholder } from './summary';
+import { TableCardProps } from './types';
+
+const defaultOnQueryChange =
+	( p: string ) =>
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Could we add details to explain why we need `any` here so that other developers can understand it quickly?

Reply:
I've looked into it one more time and decided to type it with `(path?: string, direction?: string)` instead in https://github.com/woocommerce/woocommerce/pull/36370/commits/fc819559fa8b7ba5782ff6fe63a294d88993dcee

The only place where we pass `onQueryChange` prop to `TableCard` is [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/analytics/components/report-table/index.js#L447). It uses [onQueryChange](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/navigation/src/index.js#L218) from the navigation component.

The function can return either `function(key, dir)` or `function(key)`. 

I think it's safe to use `( param: string ) => ( path?: string, direction?: string ) => {}` in this case.

Reply:
Nice sleuthing! @moon0326